icu4c/source/i18n/number_mapper.h

```diff
@@ -136,6 +136,16 @@ class AutoAffixPatternProvider {
         }
     }
 
+    inline void setTo(const AffixPatternProvider* provider, UErrorCode& status) {
+        if (auto ptr = dynamic_cast<const PropertiesAffixPatternProvider*>(provider)) {
```

Comment:
Just checking, does the variable `ptr` need to be declared outside of the if statement?

Reply:
I think this form is allowed, as long as `ptr` is convertible to `bool`.  `nullptr` is falsy; anything else is truthy.

https://en.cppreference.com/w/cpp/language/while